code/go/0chain.net/blobbercore/challenge/protocol.go

```diff
@@ -169,53 +185,84 @@ func (cr *ChallengeEntity) LoadValidationTickets(ctx context.Context) error {
 		postData["chunk_size"] = objectPath.ChunkSize
 	}
 
+	allocMu.Unlock()
+
 	postDataBytes, err := json.Marshal(postData)
 	if err != nil {
-		Logger.Error("[db]form: " + err.Error())
+		logging.Logger.Error("[db]form: " + err.Error())
 		cr.ErrorChallenge(ctx, err)
 		return err
 	}
 	responses := make(map[string]ValidationTicket)
 	if cr.ValidationTickets == nil {
 		cr.ValidationTickets = make([]*ValidationTicket, len(cr.Validators))
 	}
+
+	accessMu := sync.Mutex{}
+	updateMapAndSlice := func(validatorID string, i int, vt *ValidationTicket) {
+		accessMu.Lock()
+		cr.ValidationTickets[i] = vt
```

Comment:
Using temp mutex to protect the access of a member variable might not thread safe, imagine if any change the `LoadValidationTickets()` is called concurrently against the same challenge object (not sure if this could happen, but once it happens, we will panic).  So if we are sure the open challenges in https://github.com/0chain/blobber/blob/ce26910d76482b50601ffa846dd7bf50a0666488/code/go/0chain.net/blobbercore/challenge/challenge.go#L127 would never duplicate, then perhaps we are safe.

Reply:
How will it panic?
It is totally in the scope of the function.

Even if it is called concurrently, it will initialize its own mutex. Don't understand how it will panic.
@peterlimg

Reply:
@peterlimg chances of same chalenge_id coming many times might be there to `validateChallenge`, but all those would be different object. will that cause panic?